framework/rendering/render_context.cpp

```diff
@@ -335,14 +366,15 @@ void RenderContext::begin_frame()
 
 		if (result == VK_SUBOPTIMAL_KHR || result == VK_ERROR_OUT_OF_DATE_KHR)
 		{
-			handle_surface_changes();
+			bool swapchain_updated = handle_surface_changes(result == VK_ERROR_OUT_OF_DATE_KHR);
 
 			result = swapchain->acquire_next_image(active_frame_index, acquired_semaphore, VK_NULL_HANDLE);
 		}
 
 		if (result != VK_SUCCESS)
 		{
 			prev_frame.reset();
+			assert(false);
```

Comment:
I'm not sure if we want to have a fixed assert in here. Wouldn't that potentially break existing error handling?